tests/golden/renameat2-noreplace

```diff
@@ -0,0 +1,2 @@
+=== renameat2 noreplace flag test
```

Comment:
(still not sure where best to review commit messages).   The commit message here is from that bad first attempt, this test is more about having two step nature of the vfs rename implementation run concurrently so that first both nodes revalidate the names and check the flags and then both nodes enter our rename call.